[vault-env] Panic when process exits cleanly

Seems there is a regression bug when running a workload with the sidecar vault in daemon mode.

After upgrading from bankvaults 1.11.5 -> 1.12.2 (and yesterday 1.13.0) we started seeing these panics in the logs for one of our cronjobs:

```
...
panic: errors: *target must be interface or implement error
26/05/2021 11:17:27 
26/05/2021 11:17:27 goroutine 1 [running]:
26/05/2021 11:17:27 errors.As(0x0, 0x0, 0x1007e20, 0xc00007cb00, 0xc00066cbd0)
26/05/2021 11:17:27 	/usr/local/go/src/errors/wrap.go:87 +0x533
26/05/2021 11:17:27 emperror.dev/errors.As(...)
26/05/2021 11:17:27 	/go/pkg/mod/emperror.dev/errors@v0.8.0/wrap_go1_13.go:26
26/05/2021 11:17:27 main.main()
26/05/2021 11:17:27 	/build/cmd/vault-env/main.go:311 +0x11c9
26/05/2021 11:17:27 panic: send on closed channel
26/05/2021 11:17:27 
26/05/2021 11:17:27 goroutine 55 [running]:
26/05/2021 11:17:27 os/signal.process(0x1212130, 0x17b4868)
26/05/2021 11:17:27 	/usr/local/go/src/os/signal/signal.go:245 +0x1a9
26/05/2021 11:17:27 os/signal.loop()
26/05/2021 11:17:27 	/usr/local/go/src/os/signal/signal_unix.go:23 +0x45
26/05/2021 11:17:27 created by os/signal.Notify.func1.1
26/05/2021 11:17:27 	/usr/local/go/src/os/signal/signal.go:151 +0x45 
```

Another log section from a bash script running with `set -e` and an istio sidecar:

```
...
2021-05-11 11:02:36	
+ curl -sSL -X POST http://127.0.0.1:15000/quitquitquit
2021-05-11 11:02:36	
OK
2021-05-11 11:02:36	
+ exit 0
2021-05-11 11:02:36	
panic: errors: *target must be interface or implement error
2021-05-11 11:02:36	
2021-05-11 11:02:36	
goroutine 1 [running]:
2021-05-11 11:02:36	
errors.As(0x0, 0x0, 0x12366c0, 0xc0003600a0, 0xc0002802a0)
2021-05-11 11:02:36	
	/usr/local/go/src/errors/wrap.go:87 +0x533
2021-05-11 11:02:36	
emperror.dev/errors.As(...)
2021-05-11 11:02:36	
	/go/pkg/mod/emperror.dev/errors@v0.8.0/wrap_go1_13.go:26
2021-05-11 11:02:36	
main.main()
2021-05-11 11:02:36	
	/build/cmd/vault-env/main.go:305 +0x112b
```

A developer also noted that he saw the same panic message when his go service crashed.

I am 99 percent sure this did not happen before, as I cannot find  anything in our centralized logging system.

Current env where we experience this issue:
Kubernetes 1.20.6
bank-vaults 1.13.0
vswh: 1.13.0
vault: 1.7.2

Example yaml on how to reproduce:

```yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: vault-env-regression
  namespace: vaultregression
spec:
  template:
    metadata:
      annotations:
        vault.security.banzaicloud.io/vault-env-daemon: 'true'
    spec:
      containers:
        - name: busybox
          image: busybox:latest
          env:
            - name: SUPER_SECRET
              value: vault:secrets/data/somesecret#secretvalue
          # command: ['sleep', '5', '&&', 'echo', 'done']
          command: ['echo', 'done']
      restartPolicy: Never
      serviceAccountName: vaultregression
  backoffLimit: 1
  ```

/kind bug

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -16,7 +16,6 @@ jobs:
       K8S_VERSION: v1.19.2
       HELM_VERSION: v3.3.1
       KUBECONFIG: /home/circleci/.kube/config
-      KURUN_VERSION: "0.4.1"
       KIND_VERSION: v0.8.1
       GOPATH: /go
       DOCKER_LATEST: 1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -70,7 +70,9 @@ jobs:
       uses: actions/checkout@v2
 
     - name: Start kind cluster
-      run: kind create cluster --config hack/kind.yaml
+      run: |
+        kind version
+        kind create cluster --config hack/kind.yaml
 
     - name: Build Docker images
       run: |

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -55,7 +55,7 @@ jobs:
     name: Acceptance test
     runs-on: ubuntu-latest
     env:
-      KURUN_VERSION: "0.4.1"
+      KURUN_VERSION: "0.5.3"
       DOCKER_LATEST: 1
       VERSION: latest
 

--- a/cmd/vault-env/main.go
+++ b/cmd/vault-env/main.go
@@ -307,12 +307,15 @@ func main() {
 
 		close(sigs)
 
-		var eerr exec.ExitError
-		if errors.As(err, &eerr) {
-			os.Exit(cmd.ProcessState.ExitCode())
-		} else if err != nil {
-			logger.Fatalln("failed to exec process", entrypointCmd, err.Error())
-			os.Exit(-1)
+		if err != nil {
+			exitCode := -1
+			// try to get the original exit code if possible
+			var exitError exec.ExitError
+			if errors.As(err, &exitError) {
+				exitCode = exitError.ExitCode()
+			}
+			logger.Errorln("failed to exec process", entrypointCmd, err.Error())
+			os.Exit(exitCode)
 		} else {
 			os.Exit(cmd.ProcessState.ExitCode())
 		}

